scripts/split-e2e-tests-codebuild.ts

```diff
@@ -0,0 +1,341 @@
+    for (let test of testSuites) {
+      const currentJob = osJobs[osJobs.length - 1];
+
+      // if the current test is excluded from this OS, skip it
+      if (TEST_EXCLUSIONS[os].find((excluded) => test === excluded)) {
+        continue;
+      }
+      const FORCE_REGION = FORCE_REGION_MAP.get(test);
+      const USE_PARENT = USE_PARENT_ACCOUNT.some((usesParent) => test.startsWith(usesParent));
+
+      if (isMigration || RUN_SOLO.find((solo) => test === solo)) {
+        const newSoloJob = createRandomJob(os);
+        newSoloJob.tests.push(test);
+        if (FORCE_REGION) {
+          newSoloJob.region = FORCE_REGION;
+        }
+        if (USE_PARENT) {
+          newSoloJob.useParentAccount = true;
+        }
+        soloJobs.push(newSoloJob);
+        continue;
+      }
+
+      // add the test
+      currentJob.tests.push(test);
+      if (FORCE_REGION) {
+        currentJob.region = FORCE_REGION;
+      }
+      if (USE_PARENT) {
+        currentJob.useParentAccount = true;
+      }
+
+      // create a new job once the current job is full;
+      if (currentJob.tests.length >= MAX_WORKERS) {
+        osJobs.push(createRandomJob(os));
+      }
+    }
+    return [...osJobs, ...soloJobs];
+  };
+  const linuxJobs = generateJobsForOS('l');
+  const windowsJobs = generateJobsForOS('w');
+  const getIdentifier = (os: string, names: string) => {
+    let jobName = `${os}_${names.replace(/-/g, '_')}`.substring(0, 127);
+    if (isMigration) {
+      const startIndex = baseJobLinux.identifier.lastIndexOf('_');
+      jobName = jobName + baseJobLinux.identifier.substring(startIndex);
+    }
+    return jobName;
+  };
+  const result: any[] = [];
+  linuxJobs.forEach((j) => {
+    if (j.tests.length !== 0) {
+      const names = j.tests.map((tn) => getOldJobNameWithoutSuffixes(tn)).join('_');
+      const tmp = {
+        ...JSON.parse(JSON.stringify(baseJobLinux)), // deep clone base job
+        identifier: getIdentifier(j.os, names),
+      };
+      tmp.env.variables = {};
+      tmp.env.variables.TEST_SUITE = j.tests.join('|');
+      tmp.env.variables.CLI_REGION = j.region;
+      if (j.useParentAccount) {
+        tmp.env.variables.USE_PARENT_ACCOUNT = 1;
+      }
+      result.push(tmp);
+    }
+  });
+  windowsJobs.forEach((j) => {
+    if (j.tests.length !== 0) {
+      const names = j.tests.map((tn) => getOldJobNameWithoutSuffixes(tn)).join('_');
+      const tmp = {
+        ...JSON.parse(JSON.stringify(baseJobWindows)), // deep clone base job
+        identifier: getIdentifier(j.os, names),
+      };
+      tmp.env.variables = {};
+      tmp.env.variables.TEST_SUITE = j.tests.join('|');
+      tmp.env.variables.CLI_REGION = j.region;
+      tmp.env.variables.USE_PARENT_ACCOUNT = j.useParentAccount;
+      result.push(tmp);
+    }
+  });
+  return result;
+};
+function main(): void {
+  const configBase: any = loadConfigBase();
+  const baseBuildGraph = configBase.batch['build-graph'];
+
+  const counts = { w: 0, l: 0 };
```

Comment:
what is this variable for?

Reply:
leftover from original test splitting logic, going to remove it; thanks!